### ISSUE TITLE
Option to not do post-job cleanup

### ISSUE THREAD

**Issue by Piedone** [actions/checkout#1149]
The action does some cleanup in post-job (starting with the message "Post job cleanup."). I guess this is useful if the runner is reused, but I don't see the point of doing any cleanup on an ephemeral runner, that will be thrown away anyway. This is an issue because this cleanup can actually take significant time, like a minute, as you can also see [in this run](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/4048900879/jobs/6964683274#step:20:1). So, it would be great to be able to turn this feature off.

I'm aware that the `clean` parameter exists, but that's for cleaning in the beginning, before fetching.

**Comment by wlgrd** [comment]
I agree. I also have a use case where I need to debug the runner after execution and need the checked-out code to persist.

**Comment by peter-miroshnikov** [comment]
Same here, i execute other cleanups before that and it will be weird to exclude only the files checked out in order to delete them later with this action.

**Comment by AdamJudge** [comment]
We're having the post-job cleanup cause occasional errors, so yes if it's possible to disable this it would be great

**Comment by line-joonsoo** [comment]
I think this is essential, because I having an issue with post-job cleanup...
Here is setup:
Runner.yml
```batchfile
    steps:
      - name: Checkout action scheduler
        uses: actions/checkout@v3
      - name: Schedule automerge
        uses: ./.github/runner/auto_merge_runner  # The action.yml file is contained in this path of github actions running repo
``` 

actions.yml
```batchfile
  steps:
    - uses: actions/checkout@v3
    - name: Checkout Working Repo
      uses: actions/checkout@v3
      with:
        repository: 'some/repository'
        ref: feature/github_action
        token: ${{ inputs.gh_token }}
     ...
     Working steps
     ...
``` 

And then, it works well on each process steps, but fails on "Post Schedule automerge" steps:
```
Error: Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under '/xxx/xxx/_work/xxx/xxx/.github/runner/auto_merge_runner
``` 

Because, it's already checked out to "some/repository", the `actions.yml` file can't be found which is on original github actions running repo.
So I think it needs to reverse-checkout before doing post action, or just give options to disable post action

**Comment by snps-midhunn** [comment]
I believe this will be an important feature to have for re-using workspace across jobs. If the build artifacts are of significant size and a regression has to be run on it we can use the same workspace.

**Comment by Kristiansmp** [comment]
We would like to have this feature also. We use kota65535/github-openvpn-connect-action to create VPN connection.
When the Post Connect to VPN is triggered, it causes the network to change (NETWORK CHANGED). Only way to control this is to Connect to VPN in earlier phase, that is "needed" --> to trigger the Post later

**Comment by SergeyAmshaev** [comment]
- [ ] Sergey0021@gmail.com

**Comment by langsmith** [comment]
Wild that this still hasn't been addressed 😕  Has anyone found a workaround?

**Comment by OleksandrKvl** [comment]
Without this functionality, it's impossible to use custom actions because of the failing "post" step.

**Comment by bsumer-wyn** [comment]
![Screen Shot 2024-05-14 at 09 17 21 AM](https://github.com/actions/checkout/assets/115162963/30d2f37d-5f63-47f7-bd1a-fb9c98bf15a6)
We are having the same issue here. Not sure if there is a solution for this.

**Comment by shadowprompt** [comment]
me too. I need remain some files for next step to use

**Comment by SriniRDev** [comment]
Is this feature available now? My workflow is failing and I need to debug the workspace for the files.

**Comment by BrendenWalker** [comment]
> Is this feature available now? My workflow is failing and I need to debug the workspace for the files.

Not as far as I know.  At this point I'd suggest copying the entire workspace to another location as a final step in the job when not a successful run.  

You could also change the checkout so that it's in a subfolder of the project workspace.  Then you can copy items of interest to a folder in the workspace which the checkout command will not cleanup.

**Comment by mankinskin** [comment]
Sometimes it may be possible to use steps instead of jobs for dependents scripts.

**Comment by tlarcher-gpsw** [comment]
Another "me too" for the stack. 

I'm reusing a bunch of actions/workflows from a central private repository, one of which uses `actions/checkout` internally. The workflow that makes use of these centralised actions/workflows runs through just fine, but fails during the post-action steps with the aforementioned error (`Error: Can't find 'action.yml', 'action.yaml' or 'Dockerfile'`).

For now, I suspect I'll need to hack around the issue with something less than ideal, but it'd be really nice to be able to throw a boolean flag and have this handled by `actions/checkout`.
